ssds/__init__.py

```diff
@@ -154,6 +165,12 @@ def _upload_local_tree(self,
                     yield ssds_key
 
     def _compose_ssds_key(self, submission_id: str, submission_name: str, path: str) -> str:
+        if path.startswith("/"):
+            path = path[1:]
+        if path.endswith("/"):
+            path = path[1:]
```

Comment:
Should this be `path = path[:-1]`?

Also, instead of:

```
        if path.startswith("/"):
            path = path[1:]
        if path.endswith("/"):
            path = path[1:]
```

Would it be better to use `path.strip('/')` here?  I think the asserts could be done away with too then.

Reply:
Yes, I think .strip is better. Thanks!